### DISCUSSION TITLE
device-onvif-camera Onvif protocol to get snapshot of hikvision camera work abnormal

### DISCUSSION THREAD

**Discussion by superrxan** [edgexfoundry/community#26] [General]
**question**
    I's testing edgex with hikvision camera's integration by device-onvif-camera service, but can not get the snapshot from camera.
I am not sure this is my camera setup issue or something else, have you ever encount this 

By the way, I have tested this functionality by happytime software and can get the snapshot sussessfully.

**Environment:**
    edgex main branch on AMD cpu

**Hikvision traffic**
And this is the http traffic between the service and camera

1. digest authentication for the camera
2. username token authentication for the camera
[traffic.zip](https://github.com/edgexfoundry/community/files/9914634/traffic.zip)

**happytime software traffic**

[snapshot_usernametoken_happytime.pcap.zip](https://github.com/edgexfoundry/community/files/9914637/snapshot_usernametoken_happytime.pcap.zip)

**Comment by superrxan** [comment]
This is the traffic using postman to get snapshot from Hikvision camera directly, this can success
[snapshot_digest.pcap.zip](https://github.com/edgexfoundry/community/files/9914684/snapshot_digest.pcap.zip)

**Reply by lenny-goodell** [reply]
@cloudxxx8 , your team validated with the Hikvision camera. 
Our team doesn't have Hikvision cameras, yet. Can you assist with this question?

**Reply by weichou1229** [reply]
@superrxan
1. The Wireshark snapshot is too large, could you just capture the specific one?
2. What is the auth mode used for Postman?
3. You can also try the auth mode "both" on the device-onvif-camera.
